src/freenet/client/async/SplitFileFetcherStorage.java

```diff
@@ -1632,7 +1650,20 @@ public boolean run(ClientContext context) {
     /** Called when a segment exits cooldown e.g. due to a request completing and becoming 
      * retryable. Must NOT be called with segment locks held. */
     public void maybeClearCooldown() {
+        jobRunner.queueNormalOrDrop(new PersistentJob() {
```

Comment:
I'm confused - why bother clearing the cooldown off the thread? Were there performance problems? Did benchmarks indicate acquiring the `cooldownLock` was a bottleneck or something?